nltk/translate/bleu_score.py

```diff
@@ -481,16 +481,14 @@ def method0(self, p_n, *args, **kwargs):
         for i, p_i in enumerate(p_n):
             if p_i.numerator != 0:
                 p_n_new.append(p_i)
-            elif _emulate_multibleu and i < 5:
-                return [sys.float_info.min]
             else:
+                # To avoid math error when math.log(0) we replace 0 with
+                # sys.float_info.min
                 _msg = str("\nCorpus/Sentence contains 0 counts of {}-gram overlaps.\n"
-                           "BLEU scores might be undesirable; "
-                           "use SmoothingFunction().").format(i+1)
+                           "BLEU score will be zero; consider "
+                           "using SmoothingFunction().").format(i+1)
                 warnings.warn(_msg)
-                # If this order of n-gram returns 0 counts, the higher order
-                # n-gram would also return 0, thus breaking the loop here.
-                break
+                p_n_new.append(sys.float_info.min)
```

Comment:
Scores SHOULDN'T be zero if they are perfect match. 

We're in a glass is half empty, half full situation and more often than not in machine translation, it's half full. Otherwise the `emulate_multibleu` option should kick in and it should reflect the canonical `multi-bleu.pl`

Reply:
I think reading #1330 and all the attached issues to that thread would better clarify the scope of BLEU.